src/main/java/lotto/domain/winning/WinningStatistics.java

```diff
@@ -11,19 +12,18 @@
 
     private Map<WinningRank, Integer> stats;
 
-    public WinningStatistics() {
-        this.stats = new TreeMap<>();
-
-        stats.put(WinningRank.FIRST, 0);
-        stats.put(WinningRank.THIRD, 0);
-        stats.put(WinningRank.FOURTH, 0);
-        stats.put(WinningRank.FIFTH, 0);
-        stats.put(WinningRank.NONE, 0);
+    private WinningStatistics(Map<WinningRank, Integer> stats) {
+        this.stats = new TreeMap<>(stats);
     }
 
-    public void add(WinningRank winningRank) {
-        int prevStats = stats.get(winningRank);
-        stats.put(winningRank, prevStats + 1);
+    public static WinningStatistics from(Map<WinningRank, Long> beforeStats) {
+        Map<WinningRank, Integer> stats = Arrays.stream(WinningRank.values())
+                .collect(toMap(Function.identity()
+                        , winningRank -> Math.toIntExact(
+                                Optional.ofNullable(beforeStats.get(winningRank))
+                                        .orElse(0L))));
```

Comment:
```suggestion
        Map<WinningRank, Integer> stats = Arrays.stream(WinningRank.values())
                .collect(toMap(Function.identity()
                        , winningRank -> Math.toIntExact(
                                Optional.ofNullable(beforeStats.get(winningRank))
                                        .orElse(0L)),
                            (count1 , count2) -> count1,
                            TreeMap::new));
```
이렇게 활용하시면 collect 할 때 부터 treeMap으로 모으실 수 있을 것 같네요 😊
count를 만드는 부분이 너무 복잡해져서 메서드로 빼면 가독성도 더 좋아질 것 같습니다 !